### PR TITLE
Cleanup Secret Manager module organization

### PR DESCRIPTION
**Pull request by dzou**
Some cleanup for secret manager module.

* Move property source integration test from autoconfigure/ to secret-manager/
* Change to using `Assert.hasText(...)`

### CODE REVIEW

File: spring-cloud-gcp-secretmanager/src/test/java/org/springframework/cloud/gcp/secretmanager/it/SecretManagerPropertySourceIntegrationTests.java [hunk 1]
```diff
@@ -101,28 +99,14 @@ public void testConfiguration() {
 		assertThat(byteArraySecret).isEqualTo("the secret data.".getBytes());
 	}
 
-	@Test
-	public void testConfigurationDisabled() {
```
Comment (meltsufin) [inline spring-cloud-gcp-secretmanager/src/test/java/org/springframework/cloud/gcp/secretmanager/it/SecretManagerPropertySourceIntegrationTests.java:105]: We're losing this test now.

File: spring-cloud-gcp-secretmanager/src/test/java/org/springframework/cloud/gcp/secretmanager/it/SecretManagerPropertySourceIntegrationTests.java [hunk 2]
```diff
@@ -49,9 +47,9 @@
 import static org.assertj.core.api.Assumptions.assumeThat;
 import static org.awaitility.Awaitility.await;
 
-public class SecretManagerIntegrationTests {
+public class SecretManagerPropertySourceIntegrationTests {
```
Comment (meltsufin) [inline spring-cloud-gcp-secretmanager/src/test/java/org/springframework/cloud/gcp/secretmanager/it/SecretManagerPropertySourceIntegrationTests.java:50]: Would it be possible to add the `@Value` test here as well like what we're doing in the other PR:

```
	@Test
	public void test() {
		try (ConfigurableApplicationContext c = applicationBuilder.run()) {
			String secret = c.getBean("secret", String.class);
			assertThat(secret).isEqualTo("hello");
		}
	}

	@Configuration
	static class TestConfiguration {

		@Value("${sm://spring-cloud-gcp-it-secret")
		private String secret;

		@Bean
		public String secret() {
			return secret;
		}
	}
```

File: spring-cloud-gcp-secretmanager/src/test/java/org/springframework/cloud/gcp/secretmanager/it/SecretManagerPropertySourceIntegrationTests.java (hunk 1 above)
Comment (dzou) [reply spring-cloud-gcp-secretmanager/src/test/java/org/springframework/cloud/gcp/secretmanager/it/SecretManagerPropertySourceIntegrationTests.java]: Gotcha - I moved this test to the configuration unit tests you wrote: SecretManagerBootstrapConfigurationTests

File: spring-cloud-gcp-secretmanager/src/test/java/org/springframework/cloud/gcp/secretmanager/it/SecretManagerPropertySourceIntegrationTests.java (hunk 2 above)
Comment (dzou) [reply spring-cloud-gcp-secretmanager/src/test/java/org/springframework/cloud/gcp/secretmanager/it/SecretManagerPropertySourceIntegrationTests.java:36]: Done.

**Comment by dzou** [comment]
I reworked the tests a bit -

I reduced the scope of the integration tests and now relying on the unit tests more since the integration tests had the problem were deleting and then recreating the secrets to clean the test environment was a flaky process. The tests to verify getting secrets for different projects and versions from the integration tests were moved to SecretManagerBootstrapConfigurationTests.java.

**Pull request merged**